R/palaeorotate.R

```diff
@@ -1,246 +1,478 @@
+#'
+#' - Seton, M., Müller, R.D., Zahirovic, S., Gaina, C., Torsvik, T.H.,
+#' Shephard, G., Talsma, A., Gurnis, M., Turner, M., Maus, S., Chandler, M.
+#' (2012). Global continental and ocean basin reconstructions since 200 Ma.
+#' Earth-Science Reviews, Volume 113, Issues 3-4, July 2012, Pages 212--270.
+#' \doi{10.1016/j.earscirev.2012.03.002}.
+#'
+#' See GPlates documentation for additional information:
+#' \url{https://gwsdoc.gplates.org/reconstruction}.
+#'
+#' @section Developer(s):
 #' Lewis A. Jones
-#' @section Auditor:
-#' Missing
+#' @section Reviewer(s):
+#' Kilian Eichenseer
+#' @importFrom utils download.file
+#' @importFrom pbapply pblapply
 #' @examples
-#' \dontrun{
 #' #Generic example with a few occurrences
-#' x <- data.frame(lng = c(2, 95, 12), lat = c(46, 12, -65), age = c(88, 203, 467))
-#' palaeorotate(x = x)
+#' occdf <- data.frame(lng = c(2, -103, -66),
+#'                 lat = c(46, 35, -7),
+#'                 age = c(88, 125, 200))
+#'
+#' #Calculate palaeocoordinates using rotation files
+#' palaeorotate(occdf = occdf, method = "grid")
 #'
-#' #Now with some real fossil occurrence data! (to be updated with internal data from Bethany/Emma)
+#' #Calculate palaeocoordinates using the GPlates API
+#' palaeorotate(occdf = occdf, method = "point")
+#'
+#' #Now with some real fossil occurrence data!
 #'
 #' #Grab some data from the Paleobiology Database
-#' x <- read.csv(paste0("https://paleobiodb.org/data1.2/colls/list.csv",
-#'                      "?base_name=Scleractinia&interval=Anisian,Piacenzian"))
+#' data(tetrapods)
 #'
 #' #Assign midpoint age of fossil occurrence data for rotation
-#' x$age <- (x$max_ma + x$min_ma)/2
+#' tetrapods$age <- (tetrapods$max_ma + tetrapods$min_ma)/2
 #'
 #' #Rotate the data
-#' palaeorotate(x = x)
+#' palaeorotate(occdf = tetrapods)
 #'
 #' #Calculate uncertainity in palaeocoordinates from models
-#' palaeorotate(x = x, uncertainty = TRUE)
-#' }
+#' palaeorotate(occdf = tetrapods, uncertainty = TRUE)
 #' @export
-palaeorotate <- function(x, model = "Merdith2021", uncertainty = FALSE) {
-    #error handling
-    if (!exists("x") | !is.data.frame(x)) {
-      stop("Please supply x as a dataframe")
+palaeorotate <- function(occdf, lng = "lng", lat = "lat", age = "age",
+                         model = "MERDITH2021", method = "grid",
+                         uncertainty = FALSE) {
+    # Error handling
+    if (!exists("occdf") || !is.data.frame(occdf)) {
+      stop("Please supply occdf as a dataframe")
     }
-    if (sum((c("lng", "lat", "age") %in% colnames(x))) != 3) {
-      stop("Column names should be: lng, lat, and age")
+
+    if (sum((c(lng, lat, age) %in% colnames(occdf))) != 3) {
+      stop("defined `lng`, `lat`, and `age` not found in `occdf`")
     }
 
-    if (sum(x$lat > 90) != 0 | sum(x$lat < -90) != 0) {
-      stop("Latitude should be more than -90 and less than 90")
+    if (any(!is.numeric(occdf[, lat]) | is.na(occdf[, lat])) ||
```

Comment:
Not 100% sure what you mean here @Buffan3369 (sorry!). This is basically doing a check to see if any latitudes are NAs or non-numerics. This implementation forces the user to remove any of these data (this way they know exactly what data are being handled).  A user may have entered their data mistakenly as non-numeric (as character strings for example), i.e. "54.5", instead of 54.5, or they may not realise they have NA values. 

In a numeric vector, NAs are interpreted as numerics. For example:

```
vec <- c(5, 2, 4, NA, 5)
is.numeric(vec[4])
is.na(vec[4])
```